### DISCUSSION TITLE
expect: body: - Multiple conditions and/or String Contains

### DISCUSSION THREAD

**Discussion by simon-lepine** [getapid/apid#199] [Q&A] [answered]
Thanks so much for building/working on apid! Should be pretty quick/easy to setup a scheduled job to run all API tests and email myself the result. I haven't found anything else that can complete that simple task...

Anyway, I'm wondering if there's a way to either 
1.) set multiple conditions for the expect: body:
I.E. something like:
expect: {
	code: 200,
	body: ["test1", "test2"]
}
2.) match part of a string in the body:
I.E. something like:
expect: {
	code: 200,
	body: "%test1%"
}

I've searched through the docs, GitHub discussions, and issues but haven't found anything yet. Maybe I missed it?

I see that it's written in Go so I looked up string.Contains() in the Go reference but I didn't find a way to make that work.

Thanks again,
Simon Le Pine

**Comment by iv-p** [comment, marked as answer]
Hi Simon!

Thank you so much for your question! Yes, there is a way to specify multiple conditions, as follows:

1) There is a way to provide boolean logic checks with the helper functions `and` and `or`, you can find more information [here](https://github.com/getapid/apid/tree/main/docs#boolean-matchers).
```
expect: {
  code: 200,
  body: and(["test1", "test2"])
}
```

2) There is a regex matcher that can do exactly that! [Here](https://github.com/getapid/apid/tree/main/docs#matchers) is a list of the matchers available at this stage.
```
expect: {
  code: 200,
  body: regex("\\w+test\\w+")
}
```

Hope this helps, if not please feel free to reach out!

**Reply by simon-lepine** [reply]
Thanks! 

I made the mistake of searching the docs for some key words (like, similar, etc) rather than reading the entire docs. I hate it when I could've found my own answer.

**Reply by iv-p** [reply]
I hope you find it useful! If you have any suggestions about features and improvements please do share!
